yabt/builders/cpp.py

```diff
@@ -42,6 +43,55 @@
 logger = make_logger(__name__)
 
 
+class CompilerConfig:
+    """Helper class for managing compiler / linker options and flags"""
+
+    def __init__(self, config, target):
+        self.compiler = self.get('compiler', config, target, 'g++')
+        self.linker = self.get('linker', config, target, 'g++')
+        self.compile_flags = self.get('compile_flags', config, target, [])
+        self.link_flags = self.get('link_flags', config, target, [])
+        self.include_path = self.get('include_path', config, target, [])
+
+    def get(self, param, config, target, fallback):
+        """Return the value of `param`, according to priority / expansion.
+
+        First priority - the target itself.
+        Second priority - the project config.
+        Third priority - a global default ("fallback").
+
+        In list-params, a '$*' term processed as "expansion term", meaning
+        it is replaced with all terms from the config-level.
+        """
+        target_val = target.props.get(param)
+        config_val = config.get(param, fallback)
+        if not target_val:
+            return config_val
+        if isinstance(target_val, list):
+            val = []
+            for el in target_val:
+                if el == '$*':
+                    val.extend(listify(config_val))
+                else:
+                    val.append(el)
+            return val
+        return target_val
+
+
+# Common C++ builder signature terms
+CPP_SIG = [
+     ('sources', PT.FileList),
+     ('in_buildenv', PT.Target),
+     ('headers', PT.FileList, None),
+     ('cmd_env', None),
+     ('compiler', PT.str, None),
+     ('linker', PT.str, None),
+     ('compile_flags', PT.list, None),
+     ('link_flags', PT.list, None),
```

Comment:
where are these defaults set ?

Reply:
at YBT level the default linker flags list is empty. at the project level, the defaults are set in YSettings, using the "settings API" that's demonstrated in the tests.

Reply:
please add some custom linker flags (like you have with the compiler flags), for the lost souls reading this.